lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1207,6 +1207,18 @@ def visit_rollup_func(self, fn, **kw):
         )
         return f"{clause} WITH ROLLUP"
 
+    def visit_string_agg_func(self, fn, **kw):
+        if len(fn.clauses) > 1:
+            clause = ", ".join(
+                elem._compiler_dispatch(self, **kw) for elem in fn.clauses[:-1]
+            )
+            return "group_concat(%s SEPARATOR %s)" % (
+                clause,
+                fn.clauses[-1]._compiler_dispatch(self, **kw),
+            )
+        else:
+            return "group_concat%s" % self.function_argspec(fn)
```

Comment:
mike do you have other suggestion to support the strange mysql syntax? https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_group-concat

Reply:
I think the code here should be clearer that string_agg is a two-argument function, which is column + delimeter.   i dont think having "comma" as an implicit default is wise here.

so the code would be:

     expr, delimeter = fn.clauses

then continue from that

Reply:
some db accept the single arg form, so sqlalchemy is not assuming anything here, it's just rendering the single arg. If the db errors that's user fault

Reply:
we're going for "platform agnostic".  so I think it should be required arg in this case

Reply:
If we use a custom name I agree. If one wants to use a custom db feature the db specific name works